src/provider/controller/ContextHandler.ts

```diff
@@ -29,20 +30,26 @@ export class ContextHandler {
     }
 
     /**
-     * Send a context to a specific app
+     * Send a context to a specific window. The promise returned may reject if provided window throws an error, so caller should take
```

Comment:
> The promise returned may reject

Assuming this is referring to the `if (window.hasContextListener())` case, rather than the `else`? Don't see how the latter could throw.

Is it strange that some failures are handled, and some are propagated up the call stack?

Reply:
It's not ideal, hence it being flagged specifically in the comment. But I'm not seeing a nicer way of handling this, given the different callers of send, without a lot of duplication